man/systemd-firstboot.xml

```diff
@@ -34,12 +34,22 @@
   <refsect1>
     <title>Description</title>
 
-    <para><command>systemd-firstboot</command> initializes basic system settings interactively during the
-    first boot, or non-interactively on an offline system image. The service is started during boot if
-    <varname>ConditionFirstBoot=yes</varname> is met, which essentially means that <filename>/etc/</filename>
-    is unpopulated, see
+    <para>The <command>systemd-firstboot.service</command> is one of the units which are used to initialize
+    the machine during "First Boot", i.e. when the system is freshly installed or after a factory reset. The
+    <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry> manager
+    itself will initialize
+    <citerefentry><refentrytitle>machine-id</refentrytitle><manvolnum>5</manvolnum></citerefentry> and preset
+    all units, enabling or disabling them according to the
+    <citerefentry><refentrytitle>systemd.preset</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+    configuration. <filename>systemd-firstboot.service</filename> is started later to interactively
```

Comment:
so i wouldn't call preset files "configuration". The result in "configuration", but the assumption is that they are shipped by the OS vendor as part of the image in /usr/, and hence "configuration" is a bit misleading I think. Just say "definitions"?

Reply:
Changed to say that "to interactively initialize basic system configuration" and "systemd.preset(5) settings".

(Though, strictly speaking, the line is blurred. Firstly, with the vconsole rework, firstboot will actually immediately apply some settings to the machine, so it's not just creating configuration, it's also initializing the machine. Secondly, presets *are* configuration. We have preset overrides in /etc after all. In fact, if the whole preset system was not *configurable*, it wouldn't make any sense. If the only source of settings for presets was packaged files, we might just as well hardcode the enablement of units in the packages. But those are details, I think that in this short description it's better to present a simplified picture where firstboot is mostly about creating some configuration files.)